ansible/playbooks/tasks/aws-cluster-bootstrap.yaml

```
@@ -0,0 +1,404 @@
+  run_once: true
+
+# The following 3 tasks determine if any of nodes will be changed by writing the the corosync file (either the file changing or not currently existing)
+# If any of the nodes will change, the `Write corosync` block will run.  It will shutdown pacemaker on all nodes, write corosync and then notify a handler to
+# start corosync.
+- name: Check if template write will make changes
+  ansible.builtin.template:  # noqa no-relative-paths
+    src: "{{ corosync_template }}"
+    dest: /etc/corosync/corosync.conf
+    owner: root
+    group: root
+    mode: '0600'
+  check_mode: true
+  register: temp_check
+
+- name: Set corosync facts
+  ansible.builtin.set_fact:
+    write_corosync: "{{ temp_check.changed }}"
+
+- name: Set corosync json data
+  ansible.builtin.set_fact:
+    corosync_data: "{{ groups.hana | map('extract', hostvars) | list | json_query('[].{host: ansible_hostname,write_corosync: write_corosync}') }}"
+  run_once: true
+
+- name: Create authkeys
+  when: authkey_data|json_query('[].cryptosum')|unique|length != 1 or authkey_data|json_query('[?isreg==`false`]')|count > 0
+  notify: Start pacemaker
+  block:
+    - name: Stop pacemaker
+      ansible.builtin.systemd:
+        name: pacemaker
+        state: stopped
+
+    - name: Create cluster secret
+      ansible.builtin.command:
+        cmd: corosync-keygen
+        creates: /etc/corosync/authkey
+      when: is_primary
+
+    - name: Fetch authkey
+      ansible.builtin.fetch:
+        src: /etc/corosync/authkey
+        dest: /tmp/authkey_tmp
+        flat: true
+      when: is_primary
+
+    - name: Copy authkey
+      ansible.builtin.copy:
+        src: /tmp/authkey_tmp
+        dest: /etc/corosync/authkey
+        owner: root
+        group: root
+        mode: '0400'
+      when: not is_primary
+
+- name: Write corosync.conf
+  when: corosync_data|json_query('[?write_corosync==`true`]')|count > 0
+  notify: Start pacemaker
+  block:
+    - name: Stop pacemaker
+      ansible.builtin.systemd:
+        name: pacemaker
+        state: stopped
+
+    - name: Write the corosync.conf template
+      ansible.builtin.template:  # noqa no-relative-paths
+        src: "{{ corosync_template }}"
+        dest: /etc/corosync/corosync.conf
+        owner: root
+        group: root
+        mode: '0600'
+
+- name: Flush handler
+  meta: flush_handlers
+
+- name: Get DefaultTasksMax value
+  ansible.builtin.command:  # noqa command-instead-of-module
+    cmd: systemctl --no-pager show
+  register: systemd_result
+  changed_when: false
+
+- name: Check max tasks infinity
+  ansible.builtin.set_fact:
+    max_tasks: infinity
+  loop: "{{ systemd_result.stdout_lines }}"
+  when: item is search("DefaultTasksMax=infinity")
+
+- name: Check max tasks integer
+  ansible.builtin.set_fact:
+    max_tasks_int: "{{ item | split('=') | last }}"
+  loop: "{{ systemd_result.stdout_lines }}"
+  when: item is search("DefaultTasksMax=") and item is not search("DefaultTasksMax=infinity")
```

This line was like this from the first beginning of this (of the original) file, but eventually we can change it to (maybe in a next PR):

```suggestion
  when:
    - item is search("DefaultTasksMax=")
    - item is not search("DefaultTasksMax=infinity")
```

---

Sure.